### PR TITLE
GitHub: testing and much better error handling

### PR DESCRIPTION
**Pull request by kevmoo**

### CODE REVIEW

**Comment by kevmoo** [comment]
I think we could release after this...

**Review by jakemac53 — APPROVED** [review]

File: mono_repo/lib/src/github_config.dart [hunk 1]
```diff
@@ -19,7 +22,59 @@ class GitHubConfig {
     Map<String, dynamic> on,
     String cron,
     this.workflows,
-  ) : on = _parseOn(on, cron);
+  ) : on = _parseOn(on, cron) {
+    if (workflows != null) {
```
Comment (natebosch) [inline mono_repo/lib/src/github_config.dart:26]: [optional] Consider flipping this with an early return to reduce indentation.

File: mono_repo/lib/src/github_config.dart [hunk 2]
```diff
@@ -19,7 +22,59 @@ class GitHubConfig {
...
+      //
+      // Validate that no workflow uses the reserved key
```
Comment (natebosch) [inline mono_repo/lib/src/github_config.dart:28]: This method is 52 lines long with 3 named sections. Consider extracting each into a named method.

File: mono_repo/test/github_command_test.dart [hunk 3]
```diff
@@ -0,0 +1,309 @@
+// Copyright (c) 2017, the Dart project authors.  Please see the AUTHORS file
```
Comment (natebosch) [inline mono_repo/test/github_command_test.dart:1]: [nit] 2020

**Review by natebosch — APPROVED** [review]

File: mono_repo/lib/src/github_config.dart (hunk 1 above)
Comment (kevmoo) [reply mono_repo/lib/src/github_config.dart:26]: ack

File: mono_repo/lib/src/github_config.dart (hunk 2 above)
Comment (kevmoo) [reply mono_repo/lib/src/github_config.dart:28]: ack

File: mono_repo/test/github_command_test.dart (hunk 3 above)
Comment (kevmoo) [reply mono_repo/test/github_command_test.dart]: fixed

**Pull request merged**